Trie/Trie/Trie.cs

```diff
@@ -0,0 +1,214 @@
+﻿// <copyright file="Trie.cs" company="Roman Levashev">
+// Copyright (c) Roman Levashev. All rights reserved.
+// Licensed under the MIT License.
+// </copyright>
+
+namespace Trie;
+
+using System;
+using System.Collections.Generic;
+using System.Xml.Linq;
+
+/// <summary>
+/// Represents a Trie (prefix tree) data structure for storing and retrieving strings efficiently.
+/// </summary>
+public class Trie
+{
+    /// <summary>
+    /// Gets the root node of the Trie.
+    /// </summary>
+    public Node Root { get;  } = new Node('\0');
+
+    /// <summary>
+    /// Adds a string to the Trie.
+    /// </summary>
+    /// <param name="element">The string to add to the Trie.</param>
+    /// <returns>
+    /// <c>true</c> if the string was successfully added;
+    /// <c>false</c> if the string already exists in the Trie.
+    /// </returns>
+    /// <exception cref="ArgumentException">
+    /// Thrown when the input string is null or empty.
+    /// </exception>
+    public bool Add(string element)
+    {
+        ArgumentNullException.ThrowIfNull(element, nameof(element));
+        if (element.Length == 0)
+        {
+            throw new ArgumentException("String cannot be empty", nameof(element));
+        }
+
+        var (endPrefixNode, nextPosition, pathStack) = this.FindLongestPrefix(element, false);
```

Comment:
pathStack дальше не нужен, можно
```suggestion
        var (endPrefixNode, nextPosition, _) = this.FindLongestPrefix(element, false);
```